examples/mongodbatlas_federated_settings_identity_provider/azure/README.md

```diff
@@ -0,0 +1,233 @@
+    Idp->>A: {keys: [{"kid": kid}, ...]}
+
+    Note right of VM: A python script (pymongo.py) runs on boot<br>It reads the MongoURI from an env-var<br>The pymongo driver gets the JWT from the metadata endpoint<br>See description below
+    VM->>Idp: http://169.254.169.254/metadata/identity/oauth2/token
+    Idp->>VM: JWT token: {<br>"iss": "https://sts.windows.net/{tenant-id}", <br>"sub": "{vm_configured_identity}"<br>}
+
+    Note right of A: A database user exist on the organization<br>username: {idp_id}/{vm_configured_identity}
+    VM ->> A: connect(JWT token, MongoURI)
+    activate A
+    A->>A: validate token and lookup database user
+    A->>VM: auth ok
+    VM->>A: insert record into database
+    A->>VM: insert record response
+    VM->>A: close connection
+    deactivate A
+```
+
+- (5) We use terraform to configure a [`user_assigned_identity`](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/user_assigned_identity) for the VM and use the [metadata endpoint](https://learn.microsoft.com/en-us/entra/identity/managed-identities-azure-resources/how-to-use-vm-token#get-a-token-using-http) to obtain the token
+  - See also [pymongo docs](https://pymongo.readthedocs.io/en/stable/examples/authentication.html#mongodb-oidc)
+  - The python script is configured by [cloud init](https://cloudinit.readthedocs.io/en/latest/reference/examples.html#writing-out-arbitrary-files) using `custom_data` on the [VM (`linux_virtual_machine`)](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/linux_virtual_machine)
+- (10) Can be configured with terraform variables:
+  - `insert_record_database`
+  - `insert_record_collection`
+  - `insert_record_fields`
+
+## Resource Table
+
+## Usage
+
+**1\. Ensure your Azure credentials are set up.**
+
+1. Install the Azure CLI by following the steps from the [official Azure documentation](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli).
+2. Run the command `az login` and this will take you to the default browser and perform the authentication.
+3. Once authenticated, it will print the user details as below:
+
+```shell
+⇒  az login
+You have logged in. Now let us find all the subscriptions to which you have access...
+The following tenants don't contain accessible subscriptions. Use 'az login --allow-no-subscriptions' to have tenant level access.
+XXXXX
+[
+  {
+    "cloudName": "AzureCloud",
+    "homeTenantId": "XXXXX",
+    "id": "XXXXX",
+    "isDefault": true,
+    "managedByTenants": [],
+    "name": "Pay-As-You-Go",
+    "state": "Enabled",
+    "tenantId": "XXXXX",
+    "user": {
+      "name": "person@domain.com",
+      "type": "user"
+    }
+  }
+]
+```
+
+**2\. Ensure your MongoDB Atlas credentials are set up.**
+
+This can be done using environment variables:
+
+```shell
+export MONGODB_ATLAS_PUBLIC_KEY="xxxx"
+export MONGODB_ATLAS_PRIVATE_KEY="xxxx"
+```
+
+**3\. TFVARS**
+
+Create a file named `vars.auto.tfvars`
```

Comment:
couldn't this file already exist in the example with dummy values?

Reply:
I think committing `vars.auto.tfvars` files is a bit dangerous 😅 
In my setup, I use a global gitignore to avoid them in all repos.
Since the user will have to change the values anyway, I don't see much added benefit of including the file.
WDYT?

Reply:
Makes sense, I also see that tfvars files are ignored in this repo.